python/doc/source/release_notes.md

```diff
@@ -0,0 +1,216 @@
+# Release notes
+
+## v0.10.0
+
+Since the 0.9.0 release, there has been 311 merged pull requests from 25 contributors.
+Below follows a summary of the biggest changes to the Python-API from these pull requests.
+In addition to the changes below, the ever-lasting quest of improving performance and squasing bugs continues.
+
+### PETSc API
+
+**Authors**: [Jørgen S. Dokken](https://github.com/jorgensd), [Francesco Ballarin](https://github.com/francesco-ballarin),
+[Jack Hale](https://github.com/jhale) and [Garth N. Wells](https://github.com/garth-wells)
+
+Mapping data between {py:class}`PETSc.Vec<petsc4py.PETSc.Vec` and {py:class}`dolfinx.fem.Function`s is now
+trivial for blocked problems by using {py:func}`dolfinx.fem.petsc.assign`. 
+
+Both solvers and assembly routines interfacing with PETSc has recieved a drastic make-over to
+improve useability and maintenance, both for developers and end-users
+
+#### Improved non-linear (Newton) solver
+
+The FEniCS project has for the last 15 years had its own implementation of a Netwon solver.
+We no longer see the need of providing this solver, as the {py:class}`PETSc SNES<petsc4py.PETSc.SNES>` solver,
+and equivalent solver for C++ provides more features than our own implementation.
+
+The previously shipped {py:class}`dolfinx.nls.petsc.NewtonSolver` is deprecated, in favor of
+{py:class}`dolfinx.fem.petsc.NonlinearProblem`, which now integrates directly with {py:class}`petsc4py.PETSc.SNES`.
+
+The non-linear problem object that was sent into {py:class}`dolfinx.nls.petsc.NewtonSolver` has been renamed
+to {py:class}`NewtonSolverNonlinearProblem<dolfinx.fem.petsc.NewtonSolverNonlinearProblem>` and is also deprecated.
+
+The new {py:class}`NonlinearProblem<dolfinx.fem.petsc.NonlinearProblem>` has additional support for blocked systems,
+such as {py:attr}`NEST<petsc4py.PETSc.Mat.Type.NEST>` by supplying `kind="nest"` to its intializer. See the documentation for further
+information.
+
+#### Improved {py:class}`dolfinx.fem.petsc.LinearProblem`
+
+- The {py:class}`dolfinx.fem.petsc.LinearProblem` now support blocked problems, either specified manually or by using
+  {py:func}`ufl.extract_blocks`. By changing the input-argument `kind`, the user can now decide if they want to use
+  the DOLFINx blocked PETSc implementation (`kind="mpi"`) or the `kind=`{py:attr}`"nest"<petsc4py.PETSc.Mat.Type.NEST>`.
+- The default behavior for non-blocked systems remains the same as before.
+- The users can now also specify a (blocked) form for preconditioning through the `P` keyword argument in the constructor.
+
+#### Assembly routines
+
+In earlier versions of DOLFINx, there were three assembly routines for {py:class}`PETSc.Vec<petsc4py.PETSc.Vec>` and {py:class}`PETSc.Mat<petsc4py.PETSc.Mat>`:
+- `assemble_*`
+- `assemble_*_block`
+- `assemble_*_nest`
+
+This was confusing to users, and causing alot of duplicate logic in codes.
```

Comment:
Best not to tell users what they found confusing - the justification to improve things is enough.